Resources/Prototypes/Chat/communication_channels.yml

```diff
@@ -0,0 +1,456 @@
+  - !type:IsGhostChatCondition
+  - !type:RangeChatCondition
+    maximumRange: 10
+  serverModifiers:
+  - !type:MainMessageChatModifier
+  - !type:EntityNameHeaderChatModifier
+  - !type:ColorFulltextChatModifier
+    defaultColorKey: Emote
+  clientModifiers:
+  - !type:SpaceMainMessageChatModifier
+  - !type:BubbleProviderChatModifier
+    speechType: Emote
+  - !type:ItalicsFulltextChatModifier
+  chatFilter:
+  - Emotes
+  chatMedium:
+  - Visual
+
+- type: communicationChannel
+  id: Dead
+  publishChatConditions:
+  - !type:IsAdminChatCondition
+  - !type:IsGhostChatCondition
+  consumeChatConditions:
+  - !type:IsAdminChatCondition
+  - !type:IsGhostChatCondition
+  serverModifiers:
+  - !type:MainMessageChatModifier
+  - !type:EntityNameHeaderChatModifier
+  - !type:ColorFulltextChatModifier
+    defaultColorKey: Dead
+  clientModifiers:
+  - !type:InsertLocChatModifier
+    targetNode: EntityNameHeader
+    locString: "DEAD: "
+    afterNode: false
+  - !type:BoldTagChatModifier
+    targetNode: EntityNameHeader
+  - !type:InsertLocChatModifier
+    targetNode: EntityNameHeader
+    locString: ": "
+  - !type:BubbleProviderChatModifier
+    speechType: Say
+  chatFilter:
+  - Dead
+  chatMedium:
+  - OOC
+
+- type: communicationChannel
+  id: LOOC
+  publishChatConditions:
+  - !type:AllChatCondition
+    subconditions:
+    - !type:IsGhostChatCondition
+      inverted: true
+    - !type:IsAboveCritChatCondition
+    - !type:HasComponentChatCondition
+      component: Sleeping
+      inverted: true
+  consumeChatConditions:
+  - !type:IsGhostChatCondition
+  - !type:RangeChatCondition
+    maximumRange: 10
+  serverModifiers:
+  - !type:MainMessageChatModifier
+  - !type:EntityNameHeaderChatModifier
+  - !type:ColorFulltextChatModifier
+    defaultColorKey: LOOC
+  clientModifiers:
+  - !type:InsertLocChatModifier
+    targetNode: EntityNameHeader
+    locString: "LOOC: "
+    afterNode: false
+  - !type:BoldTagChatModifier
+    targetNode: EntityNameHeader
+  - !type:InsertLocChatModifier
+    targetNode: EntityNameHeader
+    locString: ": "
+  - !type:BubbleProviderChatModifier
+    speechType: LOOC
+  fallbackChannels:
+  - Dead #If dead & not Moderator, this should attempt the Dead channel
+  chatFilter:
+  - LOOC
+  chatMedium:
+  - OOC
+
+- type: communicationChannel
+  id: OOC
+  publishChatConditions:
+  - !type:AllChatCondition
+    subconditions:
+    - !type:IsAdminChatCondition
+    - !type:AdminOOCEnabledChatCondition
+  - !type:OOCEnabledChatCondition
+  consumeChatConditions:
+  - !type:AllChatCondition
+  serverModifiers:
+  - !type:SessionNameHeaderChatModifier
+  - !type:ColorFulltextChatModifier
+    defaultColorKey: OOC
+  clientModifiers:
+  - !type:InsertLocChatModifier
+    targetNode: SessionNameHeader
+    locString: "OOC: "
+    afterNode: false
+  - !type:BoldTagChatModifier
+    targetNode: SessionNameHeader
+  - !type:InsertLocChatModifier
+    targetNode: SessionNameHeader
+    locString: ": "
```

Comment:
The server is responsible for formatting the location of the player name relative to the main message, yet the client is responsible for formatting all the rest. This does not make sense.

Reply:
Furthermore, this entire approach to formatting chat messages is *completely* impossible to localize.